### DISCUSSION TITLE
Support for showing pictures

### DISCUSSION THREAD

**Discussion by Walther** [helix-editor/helix#11614] [Ideas]
It would be nice if Helix could display images, where desired.

Use cases for this could include for example (but not limited to)
- viewing programmatically generated graphics in the editor, while editing the code
- previewing image assets when finding, inserting, and using filenames
- creating a pixel-based output buffer for future plugins to draw on, for richer output experience

---

There is a strong contender for a possible standard terminal graphics protocol: <https://sw.kovidgoyal.net/kitty/graphics-protocol/>

In the Kitty terminal, you can use `kitty icat image.png` to view an image directly in the terminal. In addition, the page lists 15 other applications and 9 libraries currently utilizing (and exposing) the protocol.

This is also not limited to Kitty:

> Other terminals that have implemented the graphics protocol:
> - [Konsole](https://invent.kde.org/utilities/konsole/-/merge_requests/594)
> - [wayst](https://github.com/91861/wayst)
> - [WezTerm](https://github.com/wez/wezterm/issues/986)

Other options for implementing this could potentially exist, too, but they might be behind in the adoption curve.

**Comment by kirawi** [comment]
See https://github.com/helix-editor/helix/discussions/6938

**Comment by koalazub** [comment]
Referencing the discussion on an experiment I built here: https://github.com/koalazub/helix/pull/1

That showcases what could be a conversation starter on the approach/feasibility to show pictures
